contracts/settings/ATMSettings.sol

```diff
@@ -0,0 +1,243 @@
+pragma solidity 0.5.17;
+pragma experimental ABIEncoderV2;
+
+// Libraries
+
+// Commons
+import "@openzeppelin/contracts/utils/Address.sol";
+
+// Interfaces
+import "../interfaces/SettingsInterface.sol";
+import "../atm/ATMGovernanceFactoryInterface.sol";
+import "./ATMSettingsInterface.sol";
+
+
+/**
+    @notice It manages the settings for the ATMs.
+
+    @author develop@teller.finance
+ */
+contract ATMSettings is ATMSettingsInterface {
+    using Address for address;
+
+    /** Constants */
+
+    /* State Variables */
+
+    SettingsInterface public settings;
+
+    ATMGovernanceFactoryInterface public atmGovernanceFactory;
+
+    /**
+        @notice It represents a mapping to identify whether a ATM is paused or not.
+
+        i.e.: address(ATM) => true or false.
+     */
+    mapping(address => bool) public atmToPaused;
+
+    /**
+        @notice It represents a mapping to identify the ATM used in a given market.
+        @notice A market is defined by a borrowed token and a collateral token.
+
+        i.e.: address(DAI) => address(ETH) => address(ATM Teller)
+     */
+    mapping(address => mapping(address => address)) public marketToAtm;
+
+    /** Modifiers */
+
+    /**
+        @notice It checks whether sender address has the pauser role or not.
+        @dev It throws a require error if sender hasn't the pauser role.
+     */
+    modifier withPauserRole() {
+        require(settings.hasPauserRole(msg.sender), "SENDER_HASNT_PAUSER_ROLE");
+        _;
+    }
+
+    /**
+        @notice It checks whether an address is a valid ATM address or not.
+        @dev It throws a require error if the address is not a valid ATM address.
+        @param anAddress address to test.
+     */
+    modifier withValidATM(address anAddress) {
+        require(atmGovernanceFactory.isATM(anAddress) == true, "ADDRESS_ISNT_ATM");
+        _;
+    }
+
+    /* Constructor */
+
+    constructor(address atmGovernanceFactoryAddress, address settingsAddress) public {
+        require(
+            atmGovernanceFactoryAddress != address(0x0),
+            "ATM_GOV_FACTORY_MUST_BE_PROVIDED"
+        );
+        require(settingsAddress != address(0x0), "SETTINGS_MUST_BE_PROVIDED");
+
+        atmGovernanceFactory = ATMGovernanceFactoryInterface(atmGovernanceFactoryAddress);
+        settings = SettingsInterface(settingsAddress);
+    }
+
+    /** External Functions */
+
+    /**
+        @notice It pauses an given ATM.
+        @param atmAddress ATM address to pause.
+     */
+    function pauseATM(address atmAddress)
+        external
+        withPauserRole()
+        withValidATM(atmAddress)
+    {
+        require(settings.isPaused() == false, "PLATFORM_IS_ALREADY_PAUSED");
```

Comment:
Should we allow ATMs to be un/paused separately from the platform? That way if they need to still be paused even if we unpause the platform

Reply:
Yes, it is possible. We don't want to pause an atm when the platform is paused or unpause an atm when the platform is paused. Even, when you call the isATMPaused function it verifies both (platform or atm).